### DISCUSSION TITLE
Welcome to eOS Discussions!

### DISCUSSION THREAD

**Discussion by sstaub** [sstaub/eOS#4] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by MLJ-ld** [comment]
Hello!
Thank you for your work. 
I love macro box! Can you share other project/sketch info?
Thank you. 
Giovanni Fragassi

**Comment by sstaub** [comment]
Here is an example from another user who build a small fader wing
![Faderwing](https://user-images.githubusercontent.com/6450325/126659760-f652f3ae-dfc3-4243-9b94-ce6d1d3a593f.png)

**Reply by Guiddou** [reply]
Hello, I would like to make a wing 10 fader 20 button.
I try with 6 faders but my fader in A0 sends to all the other faders at the same time and it's the same if I change pins.
Can you help me ?

**Reply by sstaub** [reply]
Did you start with the fader wing example? Maybe something with your EOS fader config is wrong. Which Microcontroller?
For further investigation I need your code.

**Reply by Guiddou** [reply]
an Arduino mega 2560. I'm a beginner, for the moment I'm using the faderwing example. I wanted to do midi, the code is perhaps simpler for me but the osc seems more suitable to me. I haven't touched the example code for the moment to test with faders connected to the Arduino.

**Reply by sstaub** [reply]
I don't think that it is the code. Is the wiring of your faders correct?

**Reply by Guiddou** [reply]
In fact it was me who made a fader connection error.
everything works correctly.
I will be able to modify my code to add faders and buttons.
thank you very much for this code

**Reply by Guiddou** [reply]
Another question, can I integrate a fader page change?

**Reply by sstaub** [reply]
Not really with this version. Handling the faders values after a page change is the problem.
There will a solution in the next version, I hope so but it needs a while.

**Reply by aurygami** [reply]
Hello @Guiddou , 
I try to do the 6 faders wing, and I have the same issue than you, when I move fader one on pin A0 it moves all the others at the same time, I try like you on pin A1 etc .. but it's the same. I try to change the wire but it change nothing, you success with this problem, so what can I do ? 
Thank you 
Aury

**Reply by Guiddou** [reply]
Hello @Aurygami
My mistake was not connecting all the faders, when all the connections were made everything worked perfectly

**Comment by MLJ-ld** [comment]
Wow!!!
Can I have sketch ?

I would like to make some buttons that send OSC messages, can you help me?

where can i find something?
Thank you
Giovanni

Il giorno gio 22 lug 2021 alle 16:51 Stefan Staub ***@***.***>
ha scritto:

> Here is an example from another user who build a small fader wing
> [image: Faderwing]
> <https://user-images.githubusercontent.com/6450325/126659760-f652f3ae-dfc3-4243-9b94-ce6d1d3a593f.png>
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/sstaub/eOS/discussions/4#discussioncomment-1038138>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATVQ36YDGQFSXXUNY6VLKJ3TZAV75ANCNFSM4YP4KO6Q>
> .
>
-- 
Giovanni Fragassi

Lighting Designer
Fotografia televisiva
ETC Eos/Augument3d Product Specialist
ETC Eos Trainer

contatti:
Tel. 334-1118368
***@***.***
***@***.***
Facebook: Giovanni Mr Long John Fragassi

**Reply by sstaub** [reply]
You find the code for the fader wing in the example folder. For making buttons have a look to the documentation.

**Reply by MLJ-ld** [reply]
Hello,

I would like to be able to send two macros, one at the push of the button and one at the release, could you give me some suggestions on the change to be made?
And if imi connect buttons with 10k resistor, pin monde isn't PULLUP right? 🤔🤔

**Reply by sstaub** [reply]
It should possible, you must modify the macro button class. Therefore you must create two messages. Try this:
Macro::Macro(uint8_t pin, uint16_t macroUp, uint16_t macroDown) {
	this->pin = pin;
	pinMode(pin, INPUT_PULLUP);
	last = digitalRead(pin);
	firePatternUp = "/eos/macro/" + String(macroUp) + "/fire";
        firePatternDown= "/eos/macro/" + String(macroDown) + "/fire";
	}

void Macro::update() {
	if ((digitalRead(pin)) != last) {
		OSCMessage fireUpdateUp(firePatternUp.c_str());
                OSCMessage fireUpdateDown(firePatternDown.c_str());
		if (last == LOW) {
			last = HIGH;
                        eos.sendOSC(fireUpdateUp);
			}
		else {
			last = LOW;
			eos.sendOSC(fireUpdateDown);
			}		
		} 
	}

And yes, if you manually take a pullup resistor only INPUT as macro is needed, but not recommanded.

**Reply by MLJ-ld** [reply]
I will modify the hardware to use the PULLUP as suggested.
where  i put the code that you sent me?
ARDUINO gives me an error on ( #include "eOS.h")..i suppose i have to download this library somewhere?
i'm trying to make a macro buttons to shoot the CUEs and the submasters (the subs will shoot them without macros I think ..) i'm using arduino Mega..
Thank you for help.![image](https://user-images.githubusercontent.com/82513403/128417946-4a6582ec-8e5e-4178-991a-a41c00ad735c.jpeg)

**Reply by sstaub** [reply]
1. You need no external pull-up resistor
2.  You  can download the library here from the repository
3. For firing cues there is an example

**Reply by MLJ-ld** [reply]
1. Sorry...yes, I expressed myself badly.
2. ok! I'm new to this experience, forgive my embarrassment
3. I would like to shoot cue when the button is pressed, and when the button is released I would like go to cue 0 of that cue list; I was thinking of doing it with macros, so I asked you for information on pressing and releasing the button.
4. the code that you kindly wrote me, should it be copied in the viod setup?
5. my idea is to afre an "add-on" for console and nomad, where with the buttons I shoot cue and subs, and with the small switc I shoot macro with snapshots in the background inside. On the consoles the push-button panel fits onto the wheel section of the rate (note the shape of the pcb) what do you think?

Thank you for all.

**Reply by sstaub** [reply]
5. Maybe that Stream Deck with Companion is a better solution.

**Comment by MLJ-ld** [comment]
Ok perfect thank you!!
With example I can modify sketch and wiring!
Thank you!
Giovanni
Il giorno gio 22 lug 2021 alle 17:11 Stefan Staub ***@***.***>
ha scritto:

> You find the code for the fader wing in the example folder. For making
> buttons have a look to the documentation.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/sstaub/eOS/discussions/4#discussioncomment-1038241>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATVQ365RBVUJQ7NNJOEYFATTZAYIJANCNFSM4YP4KO6Q>
> .
>
-- 
Giovanni Fragassi

Lighting Designer
Fotografia televisiva
ETC Eos/Augument3d Product Specialist
ETC Eos Trainer

contatti:
Tel. 334-1118368
***@***.***
***@***.***
Facebook: Giovanni Mr Long John Fragassi

**Comment by skafrenzy** [comment]
Hi, love the mods you've made it the original box.

Was wondering if there was a way to 'scale' the encoders like the original sketch?

**Comment by sstaub** [comment]
did you mean `#define WHEEL_ACC		1` in line 54 of eOS.h ?

**Reply by skafrenzy** [reply]
yes, works like a charm
thank you for your time

**Comment by mikeyhill88** [comment]
> ## 👋 Welcome!
> We’re using Discussions as a place to connect with other members of our community. We hope that you:
> 
> * Ask questions you’re wondering about.
> * Share ideas.
> * Engage with other community members.
> * Welcome others and are open-minded. Remember that this is a community we
>   build together 💪.
> 
> To get started, comment below with an introduction of yourself and tell us about what you do with this community.

Hello I am a New York Based EOS programmer working in TV and Film.

Do you know how I could take the Fader Values and Labels from OSC  and print them to a small LCD Screen.

I have been messing around with the pan and tilt project  lighthack box , and but would like to substitute the Pan and Tilt for Fader Values.

Thanks
Michael

**Reply by sstaub** [reply]
This should possible. Have a look to the diagnostic tab 99, what EOS return when changing fader values. This could handle by the OSC library.

**Comment by mikeyhill88** [comment]
I looked in  the diagnostic tab 99 at what the faders return. I attached a screen shot. My question is now how do I turn this info into code I can put in my Arduiono so it can output to an LCD. I am having trouble wrapping my head around this concept. ThanksMichael   On Mon, Feb 6, 2023, 1:15 PM Stefan Staub ***@***.***> wrote:This should possible. Have a look to the diagnostic tab 99, what EOS return when changing fader values. This could handle by the OSC library.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

**Comment by Guiddou** [comment]
Hello staub,
First of all thank you for everything.
I would like to make a box2A with 4 encoders.
I'm not sure my code is good, I have the 4 encoders working but sometimes the box crashes.
I think I made a mistake.
THANKS

**Reply by sstaub** [reply]
Which board? I don't think this will work with an Arduino UNO, you need a MEGA because of RAM size. Or try a Raspberry Pi Pico.

**Reply by Guiddou** [reply]
I use in arduino mega

**Reply by sstaub** [reply]
Should work, have a look to the Diagnostic Tab 99 to see what happens.

**Reply by Guiddou** [reply]
Thanks, I'll start from the beginning and see if I make a mistake along the way.

**Reply by Guiddou** [reply]
![IMG_20241015_093849(1)](https://github.com/user-attachments/assets/2a4525b1-1a60-4c9c-8f7d-9ce8ef7ec79e)
MERCI

**Comment by djn1111** [comment]
Hello Stefan,

I am enjoying your work for quit some time with a 10 fader 2 x 10 buttons box,. using a Arduino Micro and 2x MCP chips. (if you like a picture let me know).

For quit some time I am thinking it might be nice to have a display with fader information. I started out wit looking at the que box. And to be honest some bits for example the parser are a bit over my head. But with extracting what I don't need. Looking and reading and learning I got it to work close to what I want. The problem is at the moment. It works for two faders, but when I try to add two more, the info is not displayed at start-up. When I change the function in EOS, the info is displayed so I think the problem is in the setup. Other thing more difficult I think, all fader info works close to what I want, but not if I make it a cue list, in that case it gets a bit messy.
tha has probably something to do with the Parser. I hope you have some time to point me in the right direction.

Kind regards DJN

**Reply by sstaub** [reply]
Without the code it is not clear what you want to achieve. Maybe the Arduino Micro is a part of the problem.
Also the RAM size can be a problem, therefore I'm going away from the AVRs to RaspberryPiPico.

**Reply by djn1111** [reply]
Hello, sorry I am quit new at Github, what is the best way to post the code. For this project I am testing on a Nano.

**Reply by sstaub** [reply]
Try to copy the ino file directly in your post. If doesn't work zip your ino file before copy in the post.

**Comment by djn1111** [comment]
[faderdisplayinfo2_3.ino.zip](https://github.com/user-attachments/files/21167858/faderdisplayinfo2_3.ino.zip)

I hope this works

**Reply by sstaub** [reply]
Try following things:
- after line 269 `SLIPSerial.begin(115200);` make a delay of 2000ms
- maybe delete the lines 272 through 278

It seems this is a timing problem with the Arduino Nano

**Reply by sstaub** [reply]
What says the diagnostic tab (99) with OSC settings on when you attach the Nano?

**Reply by djn1111** [reply]
With the 2000ms delay it won't start, there is no hand shake. without it starts normal. wit a reset it will start.

When te lines ar deleted, works starts op normal.

Display issue is the same still only two at startup, displayed when changed.

I made some screen shots of tab 99

[tab99 screen shots.zip](https://github.com/user-attachments/files/21169024/tab99.screen.shots.zip)

**Reply by sstaub** [reply]
In the screenshots of the display I can see that "Sub 1" is twice?! 
The connections and OSC messages in tab99 looks okay. Does the application stops working after a while?
I think there is a problem with 'String' handling in general. You should delete the data in your structure first before you override it.

**Reply by djn1111** [reply]
Hello,

Yes it was looking okay by me to. but you can make a fader a sub grand master etc. so you can make 99 sub 1 if you want to not really handy but it is possible. sometimes it is handy to have a certain sub on every fader page. Conclusion I had two sub 1's :-)

No it does not stop working after a wile, although I have not let it run for a while, I will
Are you thinking of buffer overload

**Reply by djn1111** [reply]
Could the speed be problem ? Usb speed versus read string, edit it, etc, next string ?

**Reply by sstaub** [reply]
Your tab99 screenshot says that fader 2 is Sub 2 L201
No USB speed problem.
Strings are very problematic on small microcontrollers and can causes heap overflows.

**Reply by djn1111** [reply]
Hello, sub 2 with label L201  (Lee color 201) that is oke, as described above. I will leave it on for a few hours to check overflow. Could the LCD display lib be the problem ? 
But no obvious problems in the code ?

Will try to port it to a pi pico

**Reply by djn1111** [reply]
If you look a the pictures tbelow here is  S 1 L136 2x

**Reply by sstaub** [reply]
I think the problem is the String handling in your parser.

**Reply by djn1111** [reply]
I did some more testing. what's hot en what's not :-)

Not Working:
-1-  If it start it won't show the whole display, only fader 1 & 2, after changes in fader 3 & 4 they got displayed
-2- If it goes in to idle mode (from connection to no connection with EOS) it won't start up again. if I change something in the fader      setup it give's some crazy characters, needs a reset to work again.

Working:
-3- The things that are displayed are correct, tried several things they all work out, even the Cue List is working correctly  :-)
-4-  Adding a physical fader also works, but I think it influence the start up, if the input float it will not start. I left it out for now to be more clear what is happening.

So I think not to bad, but my C++ knowledge is too low to solve this problems. But a first step is maybe to leave out the Parser, and just print the four Strings on the display one by one. If that works we now It is in the bit we took out. And go from there.
Not working "2" might suggest a oferflow.

**Comment by djn1111** [comment]
![Fader bank](https://github.com/user-attachments/assets/d30b9097-5a59-419c-8207-21b8fb3ab757)

This is with me for the past four years :-) with small performances.

Display project::
![Start up display](https://github.com/user-attachments/assets/e340fb96-cbb9-4f52-8bb6-f0b3c80f47ce)

![After startup connected to nomad](https://github.com/user-attachments/assets/45993ed6-ad23-42b1-a85a-d3464aac9986)

![After changing the fader function](https://github.com/user-attachments/assets/34c62699-69b7-4ac6-ad40-1bbd20d3f372)

And some visual info

**Reply by macrod911** [reply]
Hiya

Sorry for the delay in sending this - I've attached a zip file of my sketch that includes the Channel Number!

I have this running on an Arduino Nano Every.

I've also been playing around with your Faders code. I get the same result as you (running on the Nano Every) - only two Fader Names displaying unless the Faders are activated but I've got a couple of ideas that I'll try out as soon as I can.

[ETC_Nomad_Box_v3.zip](https://github.com/user-attachments/files/22783573/ETC_Nomad_Box_v3.zip)

**Comment by sstaub** [comment]
The next version of the library is on the way and will solve many problems you discuss here.
Have a look to issue #14

**Reply by djn1111** [reply]
Hello,

That is really great ! I am not the biggest programmer, but if I can help with some hardware testing ?  Let me know
